### ISSUE TITLE
Null Annotation inner classes

### ISSUE THREAD

**Issue by tmrobert8** [openhab/static-code-analysis#351]
I'm sure this is related to #218 but if you have an inner class - the NotNullByDefault check shouldn't be applied.

Example:
```
@NonNullByDefault
public class Stuff {
   private class HelperClass {
   }
}
```
The helper class is marked as a warning "Class should be annotated with \@NonNullByDefault" incorrectly

**Comment by jochen314** [comment]
I am not sure, but do we just need to change in `/src/main/resources/rulesets/checkstyle/rules.xml`
```
    <module name="org.openhab.tools.analysis.checkstyle.NullAnnotationsCheck">
      <property name="severity" value="warning"/>
      <property name="checkInnerUnits" value="true"/>
    </module>
```
to
```
    <module name="org.openhab.tools.analysis.checkstyle.NullAnnotationsCheck">
      <property name="severity" value="warning"/>
      <property name="checkInnerUnits" value="false"/>
    </module>
```
for this?

**Comment by marcelrv** [comment]
This should also resolve the opposing opinions of the compiler and sat-plugin on this innerclass missing the @NonNullByDefault

**Comment by markus7017** [comment]
Did you tried to add a DTO to the class‘ filename, this triggers a different style check, which solved that problem for me. I separated those classes so regular style check is used for all others
